pkg/querier/distributor_queryable_test.go

```diff
@@ -658,6 +659,21 @@ func TestDistributorQuerier_LabelNames(t *testing.T) {
 			assert.Empty(t, warnings)
 			assert.Equal(t, labelNames, names)
 		})
+
+		t.Run("with limit", func(t *testing.T) {
```

Comment:
Can't we replicate the other sub-test, only introducing a limit?
```suggestion
		t.Run("with matchers and limit", func(t *testing.T) {
```